src/report.rs

```diff
@@ -0,0 +1,256 @@
+    response_time_maximum: usize,
+) -> ResponseMetric {
+    // Calculate percentiles in a loop.
+    let mut percentiles = Vec::new();
+    for percent in &[0.5, 0.6, 0.7, 0.8, 0.9, 0.95, 0.99, 1.0] {
+        percentiles.push(metrics::calculate_response_time_percentile(
+            response_times,
+            total_request_count,
+            response_time_minimum,
+            response_time_maximum,
+            *percent,
+        ));
+    }
+
+    // Now take the Strings out of the Vector and build a ResponseMetric object.
+    ResponseMetric {
+        method: method.to_string(),
+        name: name.to_string(),
+        percentile_50: mem::take(&mut percentiles[0]),
+        percentile_60: mem::take(&mut percentiles[1]),
+        percentile_70: mem::take(&mut percentiles[2]),
+        percentile_80: mem::take(&mut percentiles[3]),
+        percentile_90: mem::take(&mut percentiles[4]),
+        percentile_95: mem::take(&mut percentiles[5]),
+        percentile_99: mem::take(&mut percentiles[6]),
+        percentile_100: mem::take(&mut percentiles[7]),
+    }
+}
+
+pub fn request_metrics_template(metric: RequestMetric) -> String {
+    format!(
+        r#"<tr>
+        <td>{method}</td>
+        <td>{name}</td>
+        <td>{number_of_requests}</td>
+        <td>{number_of_failures}</td>
+        <td>{response_time_average}</td>
+        <td>{response_time_minimum}</td>
+        <td>{response_time_maximum}</td>
+        <td>{requests_per_second}</td>
+        <td>{failures_per_second}</td>
+    </tr>"#,
+        method = metric.method,
+        name = metric.name,
+        number_of_requests = metric.number_of_requests,
+        number_of_failures = metric.number_of_failures,
+        response_time_average = metric.response_time_average,
+        response_time_minimum = metric.response_time_minimum,
+        response_time_maximum = metric.response_time_maximum,
+        requests_per_second = metric.requests_per_second,
+        failures_per_second = metric.failures_per_second,
+    )
+}
+
+pub fn response_metrics_template(metric: ResponseMetric) -> String {
+    format!(
+        r#"<tr>
+            <td>{method}</td>
+            <td>{name}</td>
+            <td>{percentile_50}</td>
+            <td>{percentile_60}</td>
+            <td>{percentile_70}</td>
+            <td>{percentile_80}</td>
+            <td>{percentile_90}</td>
+            <td>{percentile_95}</td>
+            <td>{percentile_99}</td>
+            <td>{percentile_100}</td>
+        </tr>"#,
+        method = metric.method,
+        name = metric.name,
+        percentile_50 = metric.percentile_50,
+        percentile_60 = metric.percentile_60,
+        percentile_70 = metric.percentile_70,
+        percentile_80 = metric.percentile_80,
+        percentile_90 = metric.percentile_90,
+        percentile_95 = metric.percentile_95,
+        percentile_99 = metric.percentile_99,
+        percentile_100 = metric.percentile_100,
+    )
+}
+
+pub fn build_report(
```

Comment:
@LionsAd And this is the actual file template. It required some extra escaping (ie `{{` to escape `{`) but otherwise is the same.